### PR TITLE
Page to enter and view a treatment profile

### PR DESCRIPTION
**Pull request by jasoncalabrese**
This PR is to make it easier to bring some of the work @MilosKozak to the mainline and to prepare for adding and option to enable IOB in the mainline

### CODE REVIEW

**Comment by jasoncalabrese** [comment]
Need to add IOB tests

File: lib/websocket.js [hunk 1]
```diff
@@ -188,8 +190,19 @@ function update() {
...
+
+          profiles.list(function (err, results) {
+              // There should be only one document in the profile collection with a DIA.  If there are multiple, use the last one.
+              results.forEach(function(element, index, array) {
+                  if (element) {
+                    if (element.dia) {
+                        profileData[0] = element;
+                    }
+                  }
+              });
```
Comment (dduugg) [inline lib/websocket.js]: Based on the comment, it looks like you're accessing the last element of results using an O(n) approach. If so, consider using an O(1) implementation, such as:

```
if (results.length > 0) {
  var element = results[results.length - 1];
  if (element && element.dia) {
    profileData[0] = element;
  }
}
```
(and you may not even need the outermost `if`)

File: lib/websocket.js [hunk 2]
```diff
@@ -188,8 +190,19 @@ function update() {
...
               return treatment;
           });
-          // all done, do loadData
-          loadData( );
+
+          profiles.list(function (err, results) {
+              // There should be only one document in the profile collection with a DIA.  If there are multiple, use the last one.
+              results.forEach(function(element, index, array) {
+                  if (element) {
+                    if (element.dia) {
```
Comment (dduugg) [inline lib/websocket.js]: Lines 197-198 could be simplified to `if (element && element.dia) {`, correct?

Also, consider a standard `for` loop that goes backward from the end of `results`, and breaks when it finds `element.dia`.

File: lib/websocket.js [hunk 3]
```diff
@@ -237,6 +250,10 @@ function loadData() {
         });
     }
 
+    if (profileData) {
```
Comment (dduugg) [inline lib/websocket.js]: Is this ever false? (an empty array is truthy).

**Comment by jasoncalabrese** [comment]
@dduugg this was a quick port from the iob-cob branch, was planning on doing some cleanup, not sure if that happens here or with #452

**Pull request merged**